gap/attr.gi

```diff
@@ -2613,3 +2613,146 @@ function(D)
+  # https://www.cse.msu.edu/~cse835/Papers/Graph_connectivity_revised.pdf
+  newnetw := function(digraph, source, sink)
+    local n, mat, outn, x, y;
+    n := DigraphNrVertices(digraph);
+    mat := List([1 .. 2 * n], x -> BlistList([1 .. 2 * n], []));
+    outn := OutNeighbours(digraph);
+    for x in [1 .. DigraphNrVertices(digraph)] do
+      if x <> source and x <> sink then
+        mat[x + n][x] := true;
+      fi;
+      for y in outn[x] do
+        if x = source or x = sink then
+          mat[x][y + n] := true;
+          mat[y][x]     := true;
+        elif y = source or y = sink then
+          mat[y][x + n] := true;
+          mat[x][y]     := true;
+        else
+          mat[y][x + n] := true;
+          mat[x][y + n] := true;
+        fi;
+      od;
+    od;
+    return List(mat, x -> ListBlist([1 .. 2 * n], x));
+  end;
+
+  # The following function is an implementation of the Edmonds-Karp algorithm
+  # with some minor adjustments that take into account the fact that the
+  # capacity of all edges is 1.
+  edmondskarp := function(netw, source, sink)
+    local flow, capacity, queue, m, predecessor, edgeindex, stop, current, n, v;
+
+    flow := 0;
+    capacity := List(netw, x -> BlistList(x, x));
+    # nredges := Sum(List(netw, Length));
+
+    while true do
+      queue       := [source];
+      m           := 1;
+      predecessor := List(netw, x -> 0);
+      edgeindex   := List(netw, x -> 0);
+      stop := false;
+      while m <= Size(queue) and not stop do
+        current := queue[m];
+        n := 0;
+        for v in netw[current] do
+          n := n + 1;
+          if predecessor[v] = 0 and v <> source and capacity[current][n] then
+            predecessor[v] := current;
+            edgeindex[v]   := n;
+            Add(queue, v);
+          fi;
+          if v = sink then
+            stop := true;
+            break;
+          fi;
+        od;
+        m := m + 1;
+      od;
+
+      if predecessor[sink] <> 0 then
+        v := predecessor[sink];
+        n := edgeindex[sink];
+        while v <> 0 do
+          capacity[v][n] := false;
+          n := edgeindex[v];
+          v := predecessor[v];
+        od;
+        flow := flow + 1;
+      else
+        return flow;
+      fi;
+    od;
+  end;
+
+  # Referring once again to Abdol-Hossein Esfahanian's paper (see newnetw, above)
+  # the following lines implement Algorithm Eleven of that paper.
+  mat  := BooleanAdjacencyMatrix(digraph);
+  degs := ListWithIdenticalEntries(DigraphNrVertices(digraph), 0);
+  for i in DigraphVertices(digraph) do
+    for j in [i + 1 .. DigraphNrVertices(digraph)] do
+      if mat[i][j] or mat[j][i] then
+        degs[i] := degs[i] + 1;
+        degs[j] := degs[j] + 1;
+      fi;
+    od;
+  od;
+
+  mindegv := 0;
+  mindeg  := DigraphNrVertices(digraph) + 1;
+  for i in DigraphVertices(digraph) do
+    if degs[i] < mindeg then
+      mindeg  := degs[i];
+      mindegv := i;
+    fi;
+  od;
```

Comment:
I think mindegv:=PositionMinimum(degs) and mindeg:=degs[mindegv] should do the same and is an easier computation and easier to read.